### PR TITLE
Rest API Magento 2.4 order update overwrites existing item information

### LINKED ISSUE #22431: Rest API Magento 2.4 order update overwrites existing item information
### Preconditions (*)
magento 2.4-develop

### Steps to reproduce (*)
1. Place an order with a product with custom option.
2. On the order item in the database the custom options are stored in the `sales_order_item` table, `product_options` field in the `info_buyRequest` as "key": "value" pair.
3. Update the order using the Rest API (f.e. just hold order).

4. The `product_options` are manipulated, omitting the custom options (they are present as `options` as object, but missing in the `info_buy_request` as "key": "value" pair.

### Expected result (*)
The  `info_buyRequest` remain as they were before the API Update
image

### Actual result (*)
The  `info_buyRequest` data are changed
image

### PR DESCRIPTION
**Pull request by engcom-Charlie**
### Description (*)

Fixed and covered by tests.
### Fixed Issues (if relevant)

1. Fixes magento/magento2#22431

### Manual testing scenarios (*)

See #22431

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Comment by engcom-Charlie** [comment]
@magento run all tests

File: app/code/Magento/Sales/Model/Order/ItemRepository.php [hunk 1]
```diff
@@ -185,7 +185,9 @@ private function getItemProductOptions(OrderItemInterface $entity): array
     {
         $request = $this->getBuyRequest($entity);
         $productOptions = $entity->getProductOptions();
-        $productOptions['info_buyRequest'] = array_merge($productOptions['info_buyRequest'], $request->toArray());
+        $productOptions['info_buyRequest'] = $productOptions
```
Comment (ihor-sviziev) [inline app/code/Magento/Sales/Model/Order/ItemRepository.php]: I would also check if $productOptions['info_buyRequest'] exists. What do you thing?

Comment (engcom-Charlie) [reply app/code/Magento/Sales/Model/Order/ItemRepository.php]: I've debugged a few cases, `$productOptions` was null or an array with key `info_buyRequest` but agree that it looks not well so I'll change it.

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Review by ihor-sviziev — APPROVED** [review]

**Comment by ihor-sviziev** [comment]
Looks like failed integration test not related to changes from this PR. Approved ✅

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-7612](https://jira.corp.magento.com/browse/ENGCOM-7612) has been created to process this Pull Request

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed

**Manual testing scenario:**

1. Place an order with a product with a customizable option.
2. On the order item in the database the custom options are stored in the sales_order_item table, product_options field in the info_buyRequest as "key": "value" pair.
Screenshot from 2020-06-04 13-24-23
3. Update the order using the Rest API (f.e. just **hold** order). 
Send **POST** request **/V1/orders/{id}/hold**
4.  In the `sales_order_item` table check `custom_options` column;

**Before:** :heavy_multiplication_x: The `info_buyRequest` data are changed

Screenshot from 2020-06-04 13-26-06

**After:** :heavy_check_mark: The **info_buyRequest** remain as they were before the API Update

Screenshot from 2020-06-04 12-45-04

It was checked with Unhold, Comments, Email, Cancel, also Simple, Bundle, and Configurable products.

**Comment by engcom-Echo** [comment]
@magento run Integration Tests

**Comment by slavvka** [comment]
@magento run all tests

**Comment by slavvka** [comment]
@magento run all tests

**Pull request merged**